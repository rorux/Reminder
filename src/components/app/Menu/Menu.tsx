import * as React from 'react';
import { Link } from 'react-router-dom';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import DashboardIcon from '@mui/icons-material/Dashboard';
import PeopleIcon from '@mui/icons-material/People';
import BarChartIcon from '@mui/icons-material/BarChart';
import { ROUTES } from '@router/types';
import './style.scss';

export default function Menu() {
  return (
    <>
      <Link to={ROUTES.CALENDAR}>
        <ListItem button>
          <ListItemIcon>
            <DashboardIcon />
          </ListItemIcon>
          <ListItemText primary="Календарь" />
        </ListItem>
      </Link>
      <Link to={ROUTES.DASHBOARD}>
        <ListItem button>
          <ListItemIcon>
            <BarChartIcon />
          </ListItemIcon>
          <ListItemText primary="Напоминания" />
        </ListItem>
      </Link>
      <Link to={ROUTES.PROFILE}>
        <ListItem button>
          <ListItemIcon>
            <PeopleIcon />
          </ListItemIcon>
          <ListItemText primary="Профиль" />
        </ListItem>
      </Link>
    </>
  );
}
