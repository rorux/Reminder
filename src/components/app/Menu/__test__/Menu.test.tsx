import React from 'react';
import { Link } from 'react-router-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import Menu from '../Menu';

Enzyme.configure({ adapter: new Adapter() });

const setUp = () => shallow(<Menu />);

describe('Menu component', () => {
  let component: Enzyme.ShallowWrapper;
  beforeEach(() => {
    component = setUp();
  });
  it('Snapshot of Menu component', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render Link components', () => {
    expect(component.find(Link)).toHaveLength(3);
  });
});
