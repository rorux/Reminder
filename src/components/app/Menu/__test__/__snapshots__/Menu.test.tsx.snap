// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Snapshot of Menu component 1`] = `
<Fragment>
  <Link
    to="/calendar"
  >
    <ForwardRef(ListItem)
      button={true}
    >
      <ForwardRef(ListItemIcon)>
        <Memo(ForwardRef(DashboardIcon)) />
      </ForwardRef(ListItemIcon)>
      <ForwardRef(ListItemText)
        primary="Календарь"
      />
    </ForwardRef(ListItem)>
  </Link>
  <Link
    to="/dashboard"
  >
    <ForwardRef(ListItem)
      button={true}
    >
      <ForwardRef(ListItemIcon)>
        <Memo(ForwardRef(BarChartIcon)) />
      </ForwardRef(ListItemIcon)>
      <ForwardRef(ListItemText)
        primary="Напоминания"
      />
    </ForwardRef(ListItem)>
  </Link>
  <Link
    to="/profile"
  >
    <ForwardRef(ListItem)
      button={true}
    >
      <ForwardRef(ListItemIcon)>
        <Memo(ForwardRef(PeopleIcon)) />
      </ForwardRef(ListItemIcon)>
      <ForwardRef(ListItemText)
        primary="Профиль"
      />
    </ForwardRef(ListItem)>
  </Link>
</Fragment>
`;
